### PR TITLE
Support multiproject service mesh configs

### PR DESCRIPTION
**Pull request by elfinhe**

### CODE REVIEW

**Review by jtrbs — APPROVED** [review]

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -549,6 +552,16 @@ parse_args() {
...
         shift 2
         ;;
+      --trusted_gcp_project | --trusted-gcp-project)
+        arg_required "${@}"
+        TRUSTED_GCP_PROJECTS="${2},${TRUSTED_GCP_PROJECTS}"
+        shift 2
+        ;;
+      --environ_project_number | --environ-project-number)
+        arg_required "${@}"
+        ENVIRON_PROJECT_NUMBER="$(echo "${2}" | tr '[:upper:]' '[:lower:]')"
```
Comment (zerobfd) [inline scripts/asm-installer/install_asm:562]: A number can have upper/lowercase?

File: scripts/asm-installer/install_asm [hunk 2]
```diff
@@ -1539,6 +1552,21 @@ configure_package() {
...
+
+  if [[ "${CA}" == "mesh_ca" && -n "${TRUSTED_GCP_PROJECTS}" ]]; then
+    # Gather the trust domain aliases from projects.
+    TRUST_DOMAIN_ALIASES="${PROJECT_ID}.svc.id.goog"
+    while read -d ',' -r trusted_gcp_project; do
+      TRUST_DOMAIN_ALIASES="${TRUST_DOMAIN_ALIASES} ${trusted_gcp_project}.svc.id.goog"
+    done <<EOF
+${TRUSTED_GCP_PROJECTS}
+EOF
+    run eval kpt cfg set asm anthos.servicemesh.trustDomainAliases "${TRUST_DOMAIN_ALIASES}"
```
Comment (zerobfd) [inline scripts/asm-installer/install_asm:1564]: Why is this eval necessary?

Comment (elfinhe) [reply scripts/asm-installer/install_asm:1564]: let me add comment to explain.

Command like kpt treat the word in quotes as a single word, while kpt need ${TRUST_DOMAIN_ALIASES} to be splitting words for alias array. If we remove quotes, the lint will complain.

`eval` will translate the quoted var into words to workaround both.

File: scripts/asm-installer/install_asm (hunk 1 above)
Comment (elfinhe) [reply scripts/asm-installer/install_asm:562]: let me update it.

**Pull request merged**